apps/marginfi-v2-ui/src/components/common/Navbar/Navbar.tsx

```diff
@@ -45,18 +45,37 @@ export const Navbar: FC = () => {
     state.fetchMrgnlendState,
   ]);
 
-  const { isOraclesStale, priorityType, broadcastType, priorityFees, maxCap, maxCapType, setTransactionSettings } =
-    useUiStore((state) => ({
-      isOraclesStale: state.isOraclesStale,
-      priorityType: state.priorityType,
-      broadcastType: state.broadcastType,
-      priorityFees: state.priorityFees,
-      maxCap: state.maxCap,
-      maxCapType: state.maxCapType,
-      setTransactionSettings: state.setTransactionSettings,
-    }));
+  const {
+    isOraclesStale,
+    priorityType,
+    broadcastType,
+    priorityFees,
+    maxCap,
+    maxCapType,
+    setTransactionSettings,
+    accountLabels,
+    fetchAccountLabels,
+  } = useUiStore((state) => ({
+    isOraclesStale: state.isOraclesStale,
+    priorityType: state.priorityType,
+    broadcastType: state.broadcastType,
+    priorityFees: state.priorityFees,
+    maxCap: state.maxCap,
+    maxCapType: state.maxCapType,
+    setTransactionSettings: state.setTransactionSettings,
+    accountLabels: state.accountLabels,
+    fetchAccountLabels: state.fetchAccountLabels,
+  }));
 
   const [userPointsData] = useUserProfileStore((state) => [state.userPointsData]);
+  const [hasFetchedAccountLabels, setHasFetchedAccountLabels] = React.useState(false);
+
+  React.useEffect(() => {
```

Comment:
is there a specific reason this is being fetched in the navbar? Just wondering if it might cause any issues, especially on mobile or when switching pages

Reply:
No specific reason, we could also move this to another higher level component. Reason I did it in the navbar is cause we have the <wallet> object here. however, does indeed make more sense to do this in the _app component. Will fix that